src/main/kotlin/com/atlassian/performance/tools/report/jfr/CountingOutputStream.kt

```diff
@@ -0,0 +1,43 @@
+package com.atlassian.performance.tools.report.jfr
+
+import java.io.OutputStream
+
+/**
+ * We're not using DataOutputStream because [java.io.DataOutputStream.size] is int which will overflow with our data size.
```

Comment:
Also, we're maintaining two counters, one for absolute file position and one for relative chunk offset.
BTW I think the JFR code uses "position" as absolute (seek) and "offset" as relative (read/skip).